src/js/utils/strings.ts

```diff
@@ -36,10 +36,11 @@ export function extension(path: string): string {
         return azureFormatMatches[1];
     }
 
-    path = path.split('?')[0].split('#')[0];
-    if (path.lastIndexOf('.') > -1) {
-        return path.substr(path.lastIndexOf('.') + 1, path.length).toLowerCase();
+    const fileExtension = path.replace(/^.+?\.([\w(\-)]+)(?:[;].*)?(?:[?#].*)?$/, '$1');
```

Comment:
Why the escape and parenthesis around `-`: `(\-)`? usually in brackets you don't need to escape special characters. Also there is no need for an addition capture (parenthesis capture the matched result in vars $1, $2, etc...).

Reply:
I would argue that we should not change this pattern but have another case up top to match the `.video(mpd-time-csf)`. It's unclear how the previous code extracted a type from this as `.video(mpd-time-csf)` is not an extension supported here.

Reply:
Please use the previous pattern, and if it doesn't match then leave the old `path = path.split('?')[0].split('#')[0];` logic in there to return the old result.

Reply:
And add `'file.video(mpd-time-csf)'` to these tests.